scripts/yaml_to_cmake.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Tool for a simple parsing of YAML files and return a ;-list of <key>=<value>
+pairs to use within a CMake build file.
+'''
+
+import argparse
+import sys
+import yaml
+import pykwalify.core
+
+
+METADATA_SCHEMA = '''\
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The zephyr/module.yml file is a simple list of key value pairs to be used by
+# the build system.
+type: map
+mapping:
+  build:
+    required: true
+    type: map
+    mapping:
+      cmake:
+        required: false
+        type: str
+      kconfig:
+        required: false
+        type: str
+'''
+
+
+def main():
+    parser = argparse.ArgumentParser(description='''
+    YAML to CMake variable converter''')
+
+    parser.add_argument('-f', '--file', required=True,
+                        help='YAML file with data')
+    parser.add_argument('-s', '--section', required=True,
+                        help='Section in YAML file to parse')
+    args = parser.parse_args()
+
+    with open(args.file, 'r') as f:
+        meta = yaml.safe_load(f.read())
+
+    pykwalify.core.Core(source_data=meta,
+                        schema_data=yaml.safe_load(METADATA_SCHEMA)).validate()
+    section = meta.get(args.section)
+    if section is None:
+        sys.exit(0)
+
+    val_str = ''
+    for key in section:
+        val_str += '{}={};'.format(key, section[key])
+    print(val_str)
```

Comment:
In Zephyr it is convention to use output files instead of STDOUT.

Please align.

Reply:
After aligning, --file is ambiguous, so you need;

`-i, --input`
`-o, --output`

Reply:
will do

Reply:
> In Zephyr it is convention to use output files instead of STDOUT.

Just looked around.
Are you sure you want this output to go to a file ?
When files are written in other places, it's because they are to be used later during the build, e.g. generation of header files.
This data is supposed to be used INSIDE CMake, during configure time.
Thus writing the data to a file will result in additional code inside CMake, such as
```
FILE(READ <converted-data>)
```
instead of getting this data directly into a variable as is done now.
If think you miss the point of this code.
It's purpose is to take an existing `module.yml` and then has the content available inside CMake as a list.

I changed `--file` to `-i | --input`

Reply:
Fixed.
now using `[-i|--output]` and [-o|--output] instead.